hooks/tk-multi-publish2/basic/publish_movie.py

```diff
@@ -473,8 +737,144 @@ def _unreal_render_sequence_to_movie(self, destination_path, unreal_map_path, se
+
+        # We can't control the name of the manifest file, so we save and then rename the file.
+        _, manifest_path = unreal.MoviePipelineEditorLibrary.save_queue_to_manifest_file(queue)
+        manifest_path = os.path.abspath(manifest_path)
+        manifest_dir, manifest_file = os.path.split(manifest_path)
+        f, new_path = tempfile.mkstemp(
+            suffix=os.path.splitext(manifest_file)[1],
+            dir=manifest_dir
+        )
+        os.close(f)
+        os.replace(manifest_path, new_path)
+
+        self.logger.debug("Queue manifest saved in %s" % new_path)
+        # We now need a path local to the unreal project "Saved" folder.
+        manifest_path = new_path.replace(
+            "%s%s" % (
+                os.path.abspath(
+                    os.path.join(unreal.SystemLibrary.get_project_directory(), "Saved")
+                ),
+                os.path.sep,
+            ),
+            "",
+        )
+        self.logger.debug("Manifest short path: %s" % manifest_path)
+        # Command line parameters were retrieved by submitting a queue in Unreal Editor with
+        # a MoviePipelineNewProcessExecutor executor.
+        # https://docs.unrealengine.com/4.27/en-US/PythonAPI/class/MoviePipelineNewProcessExecutor.html?highlight=executor
+        cmd_args = [
+            sys.executable,
+            "%s" % os.path.join(
+                unreal.SystemLibrary.get_project_directory(),
+                "%s.uproject" % unreal.SystemLibrary.get_game_name(),
+            ),
+            "MoviePipelineEntryMap?game=/Script/MovieRenderPipelineCore.MoviePipelineGameMode",
+            "-game",
+            "-Multiprocess",
+            "-NoLoadingScreen",
+            "-FixedSeed",
+            "-log",
+            "-Unattended",
+            "-messaging",
+            "-SessionName=\"Publish2 Movie Render\"",
+            "-nohmd",
+            "-windowed",
+            "-ResX=1280",
+            "-ResY=720",
+            # TODO: check what these settings are
+            "-dpcvars=%s" % ",".join([
+                "sg.ViewDistanceQuality=4",
+                "sg.AntiAliasingQuality=4",
+                "sg.ShadowQuality=4",
+                "sg.PostProcessQuality=4",
+                "sg.TextureQuality=4",
+                "sg.EffectsQuality=4",
+                "sg.FoliageQuality=4",
+                "sg.ShadingQuality=4",
+                "r.TextureStreaming=0",
+                "r.ForceLOD=0",
+                "r.SkeletalMeshLODBias=-10",
+                "r.ParticleLODBias=-10",
+                "foliage.DitheredLOD=0",
+                "foliage.ForceLOD=0",
+                "r.Shadow.DistanceScale=10",
+                "r.ShadowQuality=5",
+                "r.Shadow.RadiusThreshold=0.001000",
+                "r.ViewDistanceScale=50",
+                "r.D3D12.GPUTimeout=0",
+                "a.URO.Enable=0",
+            ]),
+            "-execcmds=r.HLOD 0",
```

Comment:
This will work nicely to provide your best renders, but as you get into more complicated and larger scenes, clients will run out of gpu memory.  This should at least be configurable somewhere or anything substantial will start crashing.

Reply:
Hello, would it be ok to address your comments in another round of work? I mostly copied over what was already there, and I'm not even sure what all these settings are. We can start digging and try to improve that, but I would be happy if this first pass could be merged on a wip branch.